client/my-sites/patterns/index.web.tsx

```diff
@@ -4,33 +4,59 @@ import {
 	makeLayout,
 	redirectWithoutLocaleParamInFrontIfLoggedIn,
 	render as clientRender,
+	notFound,
 } from 'calypso/controller/index.web';
 import { PatternGalleryClient } from 'calypso/my-sites/patterns/components/pattern-gallery/client';
-import { getPatternCategorySlugs } from 'calypso/my-sites/patterns/controller';
 import { PatternsHomePage } from 'calypso/my-sites/patterns/home';
+import { getCurrentUserLocale } from 'calypso/state/current-user/selectors';
+import { getPatternCategoriesQueryOptions } from './hooks/use-pattern-categories';
 import type { RouterContext, RouterNext } from 'calypso/my-sites/patterns/types';
 
 function renderPatterns( context: RouterContext, next: RouterNext ) {
-	context.primary = (
-		<PatternsHomePage
-			category={ context.params.category }
-			isGridView={ !! context.query.grid }
-			patternGallery={ PatternGalleryClient }
-		/>
-	);
+	if ( ! context.primary ) {
+		context.primary = (
+			<PatternsHomePage
+				category={ context.params.category }
+				isGridView={ !! context.query.grid }
+				patternGallery={ PatternGalleryClient }
+			/>
+		);
+	}
```

Comment:
I've missed that during my review, what's the purpose of this `if` @fredrikekelund?

Reply:
I think it had to do with the 404 page. `context.primary` would already be populated with the 404 component, so we don't want to overwrite it (talking from memory here, so I may be missing something).